code/go/0chain.net/smartcontract/dbs/event/readmarker.go

```diff
@@ -0,0 +1,95 @@
+package event
+
+import (
+	"0chain.net/core/common"
+	"errors"
+	"fmt"
+	"gorm.io/gorm"
+)
+
+type ReadMarker struct {
+	gorm.Model
+	ClientID      string  `json:"client_id"`
+	BlobberID     string  `json:"blobber_id"`
+	AllocationID  string  `json:"allocation_id"`
+	TransactionID string  `json:"transaction_id"`
+	OwnerID       string  `json:"owner_id"`
+	Timestamp     int64   `json:"timestamp"`
+	ReadCounter   int64   `json:"read_counter"`
+	ReadSize      float64 `json:"read_size"`
+	Signature     string  `json:"signature"`
+	PayerID       string  `json:"payer_id"`
+	AuthTicket    string  `json:"auth_ticket"`
+	BlockNumber   int64   `json:"block_number"`
```

Comment:
Dont we need to define primary key with gorm.Model?

Reply:
by default ID is primary key